dapp/src/executeTask.js

```diff
@@ -19,12 +19,75 @@ async function writeTaskOutput(path, message) {
   }
 }
 
+async function processProtectedData(
+  index,
+  {
+    IEXEC_IN,
+    IEXEC_OUT,
+    appDeveloperSecret,
+    requesterSecret,
+    datasetFilename = null,
+  }
+) {
+  // Parse the protected data
+  let protectedData;
+  try {
+    const deserializerConfig = datasetFilename
+      ? { protectedDataPath: `${IEXEC_IN}/${datasetFilename}` }
+      : {};
+
+    const deserializer = new IExecDataProtectorDeserializer(deserializerConfig);
+    protectedData = {
+      chatId: await deserializer.getValue('telegram_chatId', 'string'),
+    };
+  } catch (e) {
+    throw Error(`Failed to parse ProtectedData ${index}: ${e.message}`);
+  }
+
+  // Validate the protected data
+  validateProtectedData(protectedData);
+
+  // Download and decrypt content
+  const encryptedTelegramContent = await downloadEncryptedContent(
+    requesterSecret.telegramContentMultiAddr
+  );
+
+  const telegramContent = decryptContent(
+    encryptedTelegramContent,
+    requesterSecret.telegramContentEncryptionKey
+  );
+
+  // Send telegram message
+  const response = await sendTelegram({
+    chatId: protectedData.chatId,
+    message: telegramContent,
+    botToken: appDeveloperSecret.TELEGRAM_BOT_TOKEN,
+    senderName: requesterSecret.senderName,
+  });
+
+  // Write individual result file only for single processing
+  if (index === 0) {
+    await writeTaskOutput(
+      `${IEXEC_OUT}/result.txt`,
+      JSON.stringify(response, null, 2)
+    );
+  }
```

Comment:
Move this piece of code to the `start` function (for consistency with bulk workflow)